src/main/java/com/bhoruka/bloodbank/service/CampService.java

```diff
@@ -50,4 +55,22 @@ public String createCamp(@NonNull final CreateCampRequest createCampRequest) thr
         }
         return createdCamp.getId();
     }
+
+    /**
+     * Method to get data of an existing camp.
+     *
+     * @param getCampRequest gets the request to fetch data from Dao
+     * @return CampModel object of an existing camp
+     * @throws GetCampDetailsFailedException when id returned from the dao layer is null
+     */
+    public CampModel getCamp(@NonNull final GetCampRequest getCampRequest) throws GetCampDetailsFailedException {
+        //TODO check if campId exists in DB and throw exception if not
+        Optional<CampModel> getCamp = campDao.getCamp(getCampRequest.getCampId());
+        if (getCamp.isPresent()) {
+            CampModel gotCamp = getCamp.get();
```

Comment:
this extra variable isn't adding any value. You can directly return getCamp().get()

Reply:
I was thinking about it, but I wasn't sure whether directly using getCamp.get() would bind it to a CampModel.